kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/coordinatedcommits/StorageKernelAPIAdapter.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.coordinatedcommits;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.delta.kernel.engine.coordinatedcommits.Commit;
+import io.delta.kernel.engine.coordinatedcommits.CommitResponse;
+import io.delta.kernel.engine.coordinatedcommits.GetCommitsResponse;
+import io.delta.kernel.engine.coordinatedcommits.UpdatedActions;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractCommitInfo;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractMetadata;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractProtocol;
+import io.delta.kernel.utils.FileStatus;
+
+public class StorageKernelAPIAdapter {
```

Comment:
One line comment: to convert between Kernel's coordinator classes and the equivalent classes in delta-storage.